### DISCUSSION TITLE
Management: Simplify retrying from dead letter queue

### DISCUSSION THREAD

**Discussion by bodograumann** [rabbitmq/rabbitmq-server#10940] [Other] [answered]
### Is your feature request related to a problem? Please describe.

From time to time messages get sent to a dead letter queue due to temporary outages / problems.
In this case we first analyze the error logs and usually shovel the messages back to the original queue to be processed again.
This requires manually entering the correct queue name, which is error prone and requires special care.

### Describe the solution you'd like

The management UI should suggest the correct queue to shovel to.
For example if I have a queue `incoming.queue` with `x-dead-letter-exchange` set to `""` and `x-dead-letter-routing-key` set to `"incoming.deadLetter.queue"`, the UI for the `incoming.deadLetter.queue` should suggest shoveling the messages to `incoming.queue`.

### Describe alternatives you've considered

It is probably possible to do this with the management api and a custom script, but during the analysis we are already in the UI of the right deployment, so it would be much simpler there.

### Additional context

In this case the routing from the origin queue to the dead letter queue is very simple, but I realize that this need not always be the case. Tracing it through a general exchange might be more difficult...

**Comment by michaelklishin** [comment, marked as answer]
I'm afraid it's not clear to me when exactly it should suggest that, or using what data. In general, neither the management UI nor the Shovel do not have enough context, and the UI will not have access to the information embedded into individual messages, such as their routing keys or `x-death` metadata in the headers.

"Custom scripts" sounds like a perfectly fine approach to me, that's why the management UI is built on an HTTP API available to anyone and everyone.

You are welcome to build a prototype to demonstrate what you have in mind, using the very same API.
